troubleshooting/incrementally_configuring_images.md

```diff
@@ -0,0 +1,297 @@
+   it is possible to save time by generating the first partial image with all of the successful layers.
+   Find the index of the of the failed configuration either by counting the number of successful configuration layers in the logs, or by comparing the name of the failed layer to the output of `cray cfs v3 configurations describe $CFS_CONFIGURATION_NAME`.
+   Indices for the configuration layers start at 0.
+
+## Generating a partially configured image
+
+1. (`ncn-mw#`) Set the configuration limit.
+
+   If this is the first CFS run of this procedure, this should be `0` to apply only the first layer.
+
+   ```bash
+   export CONFIGURATION_LIMIT=0
+   ```
+
+   Optionally, this can instead be set to a comma-separated list of numbers starting at 0 when trying to apply all layers up to the failed layer. e.g.
+
+      ```bash
+   export CONFIGURATION_LIMIT=0,1,2
+   ```
+
+   If this is not the first CFS run in this procedure, set this value to the next layer to be applied.
+
+   ```bash
+   export CONFIGURATION_LIMIT=<previous CONFIGURATION limit +1>
+   ```
+
+1. (`ncn-mw#`) Generate a partial configured image.
+
+    The following command will generate an image that only adds the configuration layers specified in the `CONFIGURATION_LIMIT`.
+
+   ```bash
+   cray cfs v3 sessions create --name $SESSION_NAME --configuration-name $CFS_CONFIGURATION_NAME --target-definition image --target-group Compute $BASE_IMAGE_ID --configuration-limit $CONFIGURATION_LIMIT
+   ```
+
+   If this is the last layer of the configuration, it is optionally possible to specify the name of the resulting image by adding `--target-image-map`.
+
+   ```bash
+   cray cfs v3 sessions create --name $SESSION_NAME --configuration-name $CFS_CONFIGURATION_NAME --target-definition image --target-group Compute $BASE_IMAGE_ID --configuration-limit $CONFIGURATION_LIMIT --target-image-map $BASE_IMAGE_ID <desired image name>
+   ```
+
+1. (`ncn-mw#`) Monitor CFS and retrieve the partial image id.
+
+    Monitor the CFS session with the following command until the status is complete.  If `suceeded` is true, move on to the next step, otherwise debug the failure and re-run the session as necessary.
+
+   ```bash
+   cray cfs sessions describe $SESSION_NAME --format json | jq .status.session
+   ```
+
+   Example output:
+
+   ```json
+   {
+     "completionTime": "2023-11-01T19:55:10",
+     "job": "cfs-712dee37-2b80-498c-867c-42753716cad6",
+     "startTime": "2023-11-01T19:53:09",
+     "status": "complete",
+     "succeeded": "true"
+   }
+   ```
+
+1. When the session is complete, get the resulting IMS image id.
+
+   ```bash
+   cray cfs v3 sessions describe $SESSION_NAME --format json | jq .status.artifacts
+   ```
+
+   Example output:
+
+   ```json
+   [
+     {
+       "image_id": "<IMS IMAGE ID>",
+       "result_id": "<RESULTANT IMS IMAGE ID>",
+       "type": "ims_customized_image"
+     }
+   ]
+   ```
+
+1. Prepare for the next configuration layer.
```

Comment:
This step is a little confusing to me. The way it is written, I would assume that when we are on the last layer of the configuration, then I will not be setting the `BASE_IMAGE_ID` variable to the resultant IMS image ID. But I get the impression that this should be happening either way, and the only real difference in the two cases is that if this isn't the last step, then we need to delete the CFS session and repeat the previous steps. 

If that's the case, I would probably replace this with two steps. The first one is the step that you do either way (the stuff you have in the code block, I think). And the next one then says that if it's the last layer, go to the next section, otherwise go back to the top of this section.

Does that make sense? Or am I misunderstanding the intention of this step?